PaperBlossoms/data/README.md

```diff
@@ -0,0 +1,312 @@
+
+```
+$ python3 PaperBlossoms/data/scripts/add_default_snippets_to_schema.py PaperBlossoms/data/json_schema/bonds.schema.json
+```
+
+`add_enums.py` adds or overwrites `enum` lists in JSON schemas based on entered data (primarily JSON). For example, to generate a valid list of skills and techniques for school curriculum advancements, this script will load `skills.json` and `techniques.json`, extract the names of all skills and techniques, and write those names as the `enum` list for `advance` in `curriculum` within `schools.json`. This script is intended to be used when an `enum` list needs to be refreshed (new techniques, new books, etc). It takes the following arguments:
+
+```
+$ python3 PaperBlossoms/data/scripts/add_enums.py --option books
+Writing books ...
+... to clans
+... to advantages/disadvantages
+... to armor
+... to personal effects
+... to weapons
+... to techniques
+... to schools
+... to titles
+... to bonds
+... to item patterns
+$ python3 PaperBlossoms/data/scripts/add_enums.py -h
+usage: add_enums.py [-h] [--option [{rings,clans,skills,techniques,qualities,equipment,advantages,books,resistance,currency} [{rings,clans,skills,techniques,qualities,equipment,advantages,books,resistance,currency} ...]]]
+
+Utility to add enums to json schema
+
+optional arguments:
+  -h, --help            show this help message and exit
+  --option [{rings,clans,skills,techniques,qualities,equipment,advantages,books,resistance,currency} [{rings,clans,skills,techniques,qualities,equipment,advantages,books,resistance,currency} ...]]
+                        Which enums you want to write to json schemas (defaults to all with no arguments specified)
+```
+
+### Translations
+
+Translations for the UI and the database are stored separately. In this section, I describe how translations are handled within the database.
+
+How translations are stored/added:
+
+- Database translations are stored as CSV files in the [i18n](i18n/) directory.
+- Translations are simply `string` to `string_tr` mappings.
+- All possible strings which could be translated are designated with `*_tr` fields in the database. Paper Blossoms includes a utility to generate a spreadsheet with a list strings to be translated.
+- Translators fill out the spreadsheet with translations and check it in as a new CSV.
+
+How the application fetches translations:
+
+- At application launch, Paper Blossoms automatically loads the appropriate CSV into the `i18n` table in the application database (based on the same logic ).
+- The `i18n` table is part of the view definition for each entity. Every field in the application database that could be translated has a translation field with the `*_tr` suffix, which is populated in the view definition of the entity with the appropriate `i18n.string_tr` value, or the untranslated value if a translation isn't available.
+- During runtime, the application queries the application database, fetching the `*_tr` values.
+
+## Contributing data
```

Comment:
I suspect that the Contributing Data (or an even further simplified step list, if you'd rather) should come first.  Someone opening the file wants to know "what do I need to do, in order, to submit new data".  So starting with that (and adding links or references to where to learn more about each step) feels right.